### PR TITLE
haruhi-dl: new package

### PR DESCRIPTION
**Pull request by selfisekai**
#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

**Comment by ericonr** [comment]
The commit format is `New package: pkgname-version`.

What advantage does this bring over youtube-dl? I know they claim

> This is a fork of youtube-dl, focused on bringing a fast, steady stream of updates. We'll do our best to merge patches to any site, not only youtube.

But youtube-dl already does very regular updates and all...

**Comment by selfisekai** [comment]
> What advantage does this bring over youtube-dl?

@ericonr most notable changes, as I think:
- changed license to LGPL-3.0
- dropped py2 support
- support for some websites (namely, TikTok user profiles) via [Playwright](https://playwright.dev/) browser automation (optional to install via pip; ytdl couldn't do that due to license incompatibility)
- new approach for software untied to specific domains (like PeerTube, Mastodon-like services or Funkwhale) - extractors get matched by both URL scheme and website contents, which solved possible conflicts like a one between Gab Social (`https://gab.com/ACT1TV/posts/104450493441154721`) and Facebook (`https://www.facebook.com/homokomando/posts/715625522463054`)
- various extractors improved, completely refactored or created. some of these changes were previously sent to ytdl, but never merged, some even not reviewed at all. some URLs you can compare between ytdl and hdl:
  - https://tvn24.pl/polska/edyta-gorniak-napisala-o-statystach-w-szpitalach-udajacych-chorych-na-covid-19-jerzy-polaczek-i-marek-posobkiewicz-odpowiadaja-zapraszamy-4747899
  - https://www.cda.pl/video/540114ae/vfilm
  - https://www.tiktok.com/@puczirajot
  - https://makingcents.transistor.fm/episodes/the-tech-stock-bubble
  - https://share.transistor.fm/s/e9d040c0
  - https://audycje.tokfm.pl/podcast/103073,-Same-lozka-ludzi-nie-wylecza
  - https://oko.press/fakty-i-mity-o-wilkach/
  - https://weibo.com/5720474518/JxfyRbDh6?type=repost (DASH formats)
  - https://www.rmfon.pl/podcasty/poranna-rozmowa
  - https://www.ipla.tv/wideo/serial/Miodowe-Lata/5007481/Miodowe-Lata-Mur-czyli-zemsta/6df54484ee2f09a22ff3a8b46c44f328?seasonId=5007481
  - https://podcast.midline.pl/library/tracks/17
  - https://wiadomosci.tvp.pl/50725617/dzieci-na-sprzedaz-dla-homoseksualistow (ytdl downloads the weather in the sidebar instead of the state propaganda)
  - https://abc.tvp.pl/51422403/fake-news-odc-129
  - https://ninateka.pl/film/dziwne-przygody-kota-filemona-7
  - https://player.radiozet.pl/Podcasty/Gosc-Radia-ZET/Marcelina-Zawisza-o-zmianie-definicji-gwaltu-Prawo-staje-po-stronie-oprawcy (use `--no-playlist`)
  - https://wiadomosci.onet.pl/tylko-w-onecie/milosc-w-czasach-zarazy/nbqxxwm
  - https://gospodarka.dziennik.pl/emerytury-i-ofe/artykuly/8115395,prezydent-andrzej-duda-14-emerytura-podpis-ustawa.html

**Comment by selfisekai** [comment]
@ericonr updated

File: srcpkgs/haruhi-dl/template [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+changelog="https://git.sakamoto.pl/laudompat/haruhi-dl/-/raw/master/ChangeLog"
+distfiles="https://git.sakamoto.pl/laudompat/haruhi-dl/-/archive/v${version}/haruhi-dl-v${version}.tar.gz"
+checksum=042402b20b08d84c84c876dbdc53b17db80d7d3a3969c91823f1e5f97e41daa9
+
+do_check() {
+	PYTHON=/usr/bin/python3 make offlinetest
+}
+
+post_install() {
+	rm -rf ${DESTDIR}/usr/etc
```
Comment (ericonr) [inline srcpkgs/haruhi-dl/template:23]:
```suggestion
	rm -r ${DESTDIR}/usr/etc
```

**Review by ericonr — COMMENTED** [review]
There's a bit of a struggle to see how this is sufficiently better than/different from `youtube-dl` to be packaged. The relicensing under GPL is a bit weird too.

**Pull request closed without merging**